### ISSUE TITLE
Concurrency as a limit

### ISSUE THREAD

**Issue by moshie** [SamVerschueren/listr#54]
Fantastic package! I just want to propose a feature change which involves specifying a concurrent tasks limit rather than simply passing in true.

This feature would be beneficial to large tasks as they hog memory. Limiting them into chunks saves this memory allowing large tasks to run uninterrupted.

Much like in [bluebird's map function](http://bluebirdjs.com/docs/api/promise.map.html) 

```
{ concurrency: 3 } // Limit the async chain to run 3 at a time
```

**Comment by alexpsi** [comment]
Check this PR https://github.com/SamVerschueren/listr/pull/56
Had to implement a throttle control on the amount of tasks being executed in parallel and it did its job.

**Comment by steelbrain** [comment]
Friendly bump

**Comment by SamVerschueren** [comment]
Released in `0.12.0`
